Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM perl:5.22
+
+ENV PERL_MM_USE_DEFAULT=1 PERL_CARTON_PATH=/carton
+
+COPY cpanfile cpanfile.snapshot /metacpan-web/
+WORKDIR /metacpan-web
+
+RUN cpanm App::cpm Carton && \
+    useradd -m metacpan-web -g users && \
+    mkdir /carton && \
+    cpm install -L /carton
+
+COPY . /metacpan-web
```

Comment:
I guess it depends on what the purpose of this is.  It seems like it's primarily for development, so the leaner approach seems better.  The plackup command being used is also for set up for development.  The -r option provides no real value if this isn't pointed at files that are being edited, and if you aren't using -r, you would want to use a different server like Gazelle or Starman.